components/image-builder-bob/pkg/proxy/proxy.go

```diff
@@ -79,6 +79,27 @@ func rewriteURL(u *url.URL, fromRepo, toRepo, host, tag string) {
 	u.Host = host
 }
 
+// rewriteNonDockerAPIURL is used when a url has to be rewritten but the url
+// contains a non docker api path
+func rewriteNonDockerAPIURL(u *url.URL, fromPrefix, toPrefix, host, tag string) {
```

Comment:
and unit test for this.

Reply:
Absolutely, As I mentioned in the slack thread I am waiting for a response whether we want to merge https://github.com/gitpod-io/gitpod/pull/10192 first.